enterprise/views.py

```diff
@@ -127,27 +185,52 @@ def get(self, request):
 
         data = {
             'platform_name': platform_name,
-            'sso_provider': customer.name,
+            'ec_name': customer.name,
             'data_sharing_consent': 'required' if required else 'optional',
             "messages": {
                 "warning": self.get_warning(customer.name, platform_name, required),
                 "note": self.get_note(customer.name, required),
             },
+            "course_id": None
         }
         return render_to_response('grant_data_sharing_permissions.html', data, request=request)
 
-    def post(self, request):
+    def get(self, request):
         """
-        Process the above form.
+        Render a form to collect user input about data sharing consent.
```

Comment:
I'm not sure if it should be in the docstring or in a code comment, but I think it's important to explain the choice being made here based on the presence of a course parameter and what the difference in behavior for the two branches.